sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -198,6 +203,85 @@ def is_globally_windowed(self):
             == sideinputs._global_window_mapping_fn)
 
 
+class CombiningValueRuntimeState(userstate.RuntimeState):
+  def __init__(self, underlying_bag_state, combinefn):
+    self._combinefn = combinefn
+    self._underlying_bag_state = underlying_bag_state
+
+  def _read_accumulator(self, rewrite=True):
+    merged_accumulator = self._combinefn.merge_accumulators(
+        self._underlying_bag_state.read())
+    if rewrite:
+      self._underlying_bag_state.clear()
+      self._underlying_bag_state.add(merged_accumulator)
+    return merged_accumulator
+
+  def read(self):
+    return self._combinefn.extract_output(self._read_accumulator())
+
+  def add(self, value):
+    # Prefer blind writes, but don't let them grow unboundedly.
+    if random.random() < 0.5:
```

Comment:
Can you add a rationale for this threshold or state that it is arbitrary for now?

Reply:
Done.